lib/util/utils.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import * as AsyncLock from "async-lock";
+
+export interface AsyncLockOptions {
+  /**
+   * @property {number} [timeout] The max timeout. Default is: 0 (never timeout).
+   */
+  timeout?: number;
+  /**
+   * @property {number} [maxPending] Maximum pending tasks. Default is: 1000.
+   */
+  maxPending?: number;
+  /**
+   * @property {boolean} [domainReentrant] Whether lock can reenter in the same domain. Default is: false.
+   */
+  domainReentrant?: boolean;
+  /**
+   * @property {any} [Promise] Your implementation of the promise. Default is: global promise.
+   */
+  Promise?: any;
+}
+
+export interface ParsedConnectionString {
+  Endpoint: string;
+  SharedAccessKeyName: string;
+  SharedAccessKey: string;
+  EntityPath?: string;
+  [x: string]: any;
+}
+
+export function parseConnectionString(connectionString: string): ParsedConnectionString {
+  return connectionString.split(';').reduce((acc, part) => {
+    const splitIndex = part.indexOf('=');
+    return {
+      ...acc,
+      [part.substring(0, splitIndex)]: part.substring(splitIndex + 1)
+    };
+  }, {} as any);
+}
+
+/**
+ * Gets a new instance of the async lock with desired settings.
+ * @param {AsyncLockOptions} [options] The async lock options.
+ */
+export function getNewAsyncLock(options?: AsyncLockOptions): AsyncLock {
+  return new AsyncLock(options);
+}
+
+/**
+ * @constant {AsyncLock} defaultLock The async lock instance with default settings.
+ */
+export const defaultLock: AsyncLock = new AsyncLock();
```

Comment:
I found this super useful for solving concurrency related issues. This makes sure that there is only one $cbs and $management session per connection.